File: cpp/PE036.cpp
#include <iostream>
using namespace std;
int a[100];
bool isPalindromic(int n, int base)
{
    int i = 0, j, k;
    while(n){
        a[i] = n%base;
        i++;
        n /= base;
    }
    for(j = 0, k = i-1; j < k; j++, k--){
        if(a[j] != a[k])
            return false;
    }
    return true;
}

int main()
{
    int i, sum = 0;
    for(i = 1; i < 1000000; i++){
        if(isPalindromic(i, 2) && isPalindromic(i, 10)){
            sum += i;
        }
    }
    cout << sum << endl;
    return 0;
}

File: python/PE036.py
#!/usr/bin/env python
def isPalindromic(n):
    s = str(n)
    if s == s[::-1]:
        return True
    return False

def base2(n):
    res = ""
    while n != 0:
        res = str(n%2) + res
        n /= 2
    return int(res)

def main():
    s = 0
    # A number in base 2 must be odd to be a palindrome
    for i in range(1, 1000000, 2):
        if isPalindromic(i) and isPalindromic(base2(i)):
            s += i
    print s

if __name__ == '__main__':
    main()
